pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -618,31 +621,100 @@ func (c *Cluster) saveMonConfig() error {
 var updateDeploymentAndWait = UpdateCephDeploymentAndWait
 
 func (c *Cluster) startMon(m *monConfig, hostname string) error {
-
+	// check if the monitor deployment already exists
 	d := c.makeDeployment(m, hostname)
-	logger.Debugf("Starting mon: %+v", d.Name)
-	_, err := c.context.Clientset.AppsV1().Deployments(c.Namespace).Create(d)
+	var deploymentExists bool
+	_, err := c.context.Clientset.AppsV1().Deployments(c.Namespace).Get(
+		d.Name, metav1.GetOptions{})
+	if err == nil {
+		deploymentExists = true
+	} else {
+		if errors.IsNotFound(err) {
+			deploymentExists = false
+		} else {
+			return fmt.Errorf("failed to get mon deployment %s. %+v", d.Name, err)
+		}
+	}
+
+	// if the deployment exists, also check if it is using PVC storage
+	pvc, err := c.makeDeploymentPVC(m)
 	if err != nil {
-		if !errors.IsAlreadyExists(err) {
-			return fmt.Errorf("failed to create mon deployment %s. %+v", m.ResourceName, err)
+		return fmt.Errorf("failed to make mon %s pvc. %+v", d.Name, err)
+	}
+	var pvcExists bool
+	if deploymentExists {
+		_, err = c.context.Clientset.CoreV1().PersistentVolumeClaims(
```

Comment:
Instead of checking if the PVC exists, could we look at the deployment spec retrieved above on line 627 to see if it was using a PVC?